packages/dd-trace/src/appsec/iast/analyzers/sql-injection-analyzer.js

```diff
@@ -13,6 +13,22 @@ class SqlInjectionAnalyzer extends InjectionAnalyzer {
     this.addSub('apm:mysql:query:start', ({ sql }) => this.analyze(sql, 'MYSQL'))
     this.addSub('apm:mysql2:query:start', ({ sql }) => this.analyze(sql, 'MYSQL'))
     this.addSub('apm:pg:query:start', ({ query }) => this.analyze(query.text, 'POSTGRES'))
+
+    this.addSub('datadog:sequelize:query:start', ({ sql, dialect }) => {
+      const parentStore = storage.getStore()
+      if (parentStore) {
+        this.analyze(sql, dialect.toUpperCase())
+
+        storage.enterWith({ ...parentStore, sqlAnalyzed: true, sequelizeParentStore: parentStore })
```

Comment:
This data doesn't seem to ever get cleared from the store so it may have some unintended side-effects. 🤔

Reply:
This is restored on `datadog:sequelize:query:finish`

The purpose of this lines is prevent double detections of the same vulnerability. As sequelize library uses `pg` or `mysql2` libraries, we need a way to set in store that we are already in the query. For that reason we are entering with new store in `:start` and restoring the old one in `:finish`.

Reply:
Ah, true. Do you know if sequelize could trigger these in a nested way? Setting on the store rather than storing on the span could make things leak beyond where it should. 🤔

Reply:
Good point. I've reviewed it after your comment. The answer is not, sequelize.query method does not call internally again to the same query method.
sequelize.query method calls to the query method of the select dialect (pg, sqlite, mysql...)

With sequelize we are not creating new span, so current span is the parent span, and the parent span can have multiple calls to `sequelize.query` method, that the reason that we should save the data in AsyncResource (that is new asyncresource, created in the instrumentation point) and not in the parent span.